plasmapy/simulation/abstractions.py

```diff
@@ -53,4 +57,142 @@ class AbstractTimeDependentSimulation(AbstractSimulation):
     change at any time.
     """
 
-    pass
+    ...
+
+
+class AbstractNormalizations(ABC):
+    """
+    An abstract base class to represent the normalization constants for
+    systems of equations describing plasmas.
+
+    ... warning::
+        This interface is unstable and is subject to change.
+    """
+
+    @property
+    @abstractmethod
+    def current_density(self) -> u.A / u.m ** 2:
+        """The current density normalization."""
+        ...
+
+    @property
+    @abstractmethod
+    def diffusivity(self) -> u.m ** 2 / u.s:
+        """The normalization for diffusivity."""
+        ...
+
+    @property
+    @abstractmethod
+    def dynamic_viscosity(self) -> u.Pa * u.s:
+        """The normalization for dynamic viscosity."""
+        ...
+
+    @property
+    @abstractmethod
+    def electric_field(self) -> u.V / u.m:
+        """The electric field normalization."""
+        ...
+
+    @property
+    @abstractmethod
+    def heat_flux(self) -> u.J * u.m ** -2 * u.s ** -1:
+        """The normalization for heat flux."""
+        ...
+
+    @property
+    @abstractmethod
+    def ion(self) -> Particle:
+        """
+        The ion in the plasma, which could be a positron in the case
+        of a pair plasma.
+        """
+        ...
```

Comment:
An `ion` isn't really a normalization.  It's mass could be a normalization, or it's gyrofrequency could be.  I would remove this method and leave it up to the user to manually define/calculate the other normalizations.

Reply:
Yeah...I've been wondering how to handle ions in the normalization, since how to handle ion stuff make normalization stuff kind of messy...in particular for how we handle multi-component plasmas.  Ultimately I'd like for there to be normalization classes that allow operations like:
```Python
>>> dimensionless_plasma = dimensional_plasma / normalizations
>>> dimensional_plasma = normalizations * dimensionless_plasma
```
For these to work we'd need information on the ion(s) somehow, so the choice of ion(s) do end up weirdly more or less being part of the normalization...in particular for the normalization between number density and mass density.  I'll remove the `ion` attribute for now, and worry about how to deal with this next year.

ugh why won't 2016 just end already

Reply:
> ugh why won't 2016 just end already

I know! Longest year eveeeeeeeeeer!  Completely feels like 4+ years.

---

Well, that will be an interesting implementation.  I like it, but I have no clue how to go about implementing it.  Might be easier to have something like `dimensionless_plasma.reform(normalizations)` and `dimensional_plasma.reform(normalizations)`.  I only used `reform` to keep the name consistent across all plasma classes.